xprof/xprof.rb.in

```diff
@@ -545,18 +549,20 @@ parser.on('-k', '--kernel-verbose',
 parser.on('--max-name-size SIZE',
           'Maximum size allowed for kernels names.',
           'Use -1 to mean no limit.',
-          'Default: 80') { |size| size.to_i }
+          'Default: 80') { |size| to_i_or_raise(size) }
 
 parser.on('--metadata', 'Display trace Metadata')
 parser.on('-v', '--version', 'Display THAPI version')
-parser.on('--debug [LEVEL]', 'Level of debug [default 0]') { |level| level.to_i }
+parser.on('-h', '--help', 'Display this message')
+
+parser.on('--debug [LEVEL]', 'Level of debug [default 0]') { |level| to_i_or_raise(level) }
```

Comment:
Why not:
```suggestion
parser.on('--debug [LEVEL]', OptionParser::DecimalInteger, 'Level of debug [default 0]')
```

Reply:
Oh, I never I didn't know. Far better! Let me try!